packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -52,15 +63,187 @@ export default util.createRule<Options, MessageIds>({
   defaultOptions: [
     {
       ignoreConditionalTests: true,
+      ignoreTernaryTests: true,
       ignoreMixedLogicalExpressions: true,
     },
   ],
-  create(context, [{ ignoreConditionalTests, ignoreMixedLogicalExpressions }]) {
+  create(
+    context,
+    [
+      {
+        ignoreConditionalTests,
+        ignoreTernaryTests,
+        ignoreMixedLogicalExpressions,
+      },
+    ],
+  ) {
     const parserServices = util.getParserServices(context);
     const sourceCode = context.getSourceCode();
     const checker = parserServices.program.getTypeChecker();
 
     return {
+      ConditionalExpression(node: TSESTree.ConditionalExpression): void {
+        if (ignoreTernaryTests) {
+          return;
+        }
+
+        let operator: '==' | '!=' | '===' | '!==' | undefined;
+        let nodesInsideTestExpression: TSESTree.Node[] = [];
+        if (node.test.type === AST_NODE_TYPES.BinaryExpression) {
+          nodesInsideTestExpression = [node.test.left, node.test.right];
+          if (
+            node.test.operator === '==' ||
+            node.test.operator === '!=' ||
+            node.test.operator === '===' ||
+            node.test.operator === '!=='
+          ) {
+            operator = node.test.operator;
+          }
+        } else if (
+          node.test.type === AST_NODE_TYPES.LogicalExpression &&
+          node.test.left.type === AST_NODE_TYPES.BinaryExpression &&
+          node.test.right.type === AST_NODE_TYPES.BinaryExpression
+        ) {
+          nodesInsideTestExpression = [
+            node.test.left.left,
+            node.test.left.right,
+            node.test.right.left,
+            node.test.right.right,
+          ];
+          if (node.test.operator === '||') {
+            if (
+              node.test.left.operator === '===' &&
+              node.test.right.operator === '==='
+            ) {
+              operator = '===';
+            } else if (
+              ((node.test.left.operator === '===' ||
+                node.test.right.operator === '===') &&
+                (node.test.left.operator === '==' ||
+                  node.test.right.operator === '==')) ||
+              (node.test.left.operator === '==' &&
+                node.test.right.operator === '==')
+            ) {
+              operator = '==';
+            }
+          } else if (node.test.operator === '&&') {
+            if (
+              node.test.left.operator === '!==' &&
+              node.test.right.operator === '!=='
+            ) {
+              operator = '!==';
+            } else if (
+              ((node.test.left.operator === '!==' ||
+                node.test.right.operator === '!==') &&
+                (node.test.left.operator === '!=' ||
+                  node.test.right.operator === '!=')) ||
+              (node.test.left.operator === '!=' &&
+                node.test.right.operator === '!=')
+            ) {
+              operator = '!=';
+            }
+          }
```

Comment:
The other way to solve this is to put a bunch of `else { return; }` here instead of the one `if (!operator) { return; }` at the end

Reply:
Yeah I tried refactoring a bit and couldn't find anything significantly cleaner. 🤷